### PR TITLE
feat: add unit tests for internal `_credit` function

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 1]
```diff
@@ -279,4 +289,112 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+
+        // Now it should work
+        vm.expectEmit(true, true, true, true);
+        emit IERC20.Transfer(address(0), user2, TRANSFER_AMOUNT);
+
+        uint256 amountReceived = iexecLayerZeroBridgeChainX.exposed_credit(user2, TRANSFER_AMOUNT, SOURCE_EID);
+
+        assertEq(amountReceived, TRANSFER_AMOUNT, "Amount received should equal mint amount");
+        assertEq(rlcCrosschainToken.balanceOf(user2), initialBalance + TRANSFER_AMOUNT, "User balance should increase");
+    }
```
Comment (Le-Caignec) [inline test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:399]: This test might be redundant, as the pause pattern is already thoroughly tested in the `IexecLayerZeroBridge` contract. What du you think ?

File: test/units/utils/TestUtils.sol [hunk 2]
```diff
@@ -16,35 +16,58 @@ import {Deploy as RLCCrosschainTokenDeployScript} from "../../../script/RLCCross
...
-            IexecLayerZeroBridge iexecLayerZeroBridgeChainB,
-            RLCMock rlcToken,
-            RLCCrosschainToken rlcCrosschainToken,
-            RLCLiquidityUnifier rlcLiquidityUnifier
-        )
-    {
-        address createXFactory = address(new CreateX());
+    // Struct to hold deployment parameters and reduce stack depth
+    struct DeploymentParams {
+        string iexecLayerZeroBridgeContractName;
```
Comment (Le-Caignec) [inline test/units/utils/TestUtils.sol:21]: New param Added here !

File: src/mocks/IexecLayerZeroBridgeHarness.sol
Comment (Le-Caignec) [inline src/mocks/IexecLayerZeroBridgeHarness.sol:1]: Maybe this contract shouldn’t be in the mock folder, but rather in a Harness folder at the same level of the mock folder. What do you think?

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol (hunk 1 above)
Comment (gfournieriExec) [reply test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:399]: Yes I think the 3 last tests are foccused UT on modifer worlfow rather than _credit function 
But we can keep it I don't mind

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 3]
```diff
@@ -279,4 +288,112 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
         );
     }
+
+    // ============ _credit ============
+    function test_credit_SuccessfulMintToUser() public {
+        // Test successful minting to a regular user address
+        uint256 initialBalance = rlcCrosschainToken.balanceOf(user2);
+
+        vm.expectEmit(true, true, true, true);
+        emit IERC20.Transfer(address(0), user2, TRANSFER_AMOUNT);
```
Comment (zguesmi) [inline test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:298]: we can check crosschainMint event to make sure we call the contract.

Comment (Le-Caignec) [reply test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:299]: fix Commit: 62f3ee5475761903cbba340facf6e140af0961d2

**Review by gfournieriExec — APPROVED** [review]
LGTM thanks 🚢

File: src/mocks/IexecLayerZeroBridgeHarness.sol
Comment (zguesmi) [inline src/mocks/IexecLayerZeroBridgeHarness.sol:1]: Let's put all mocks inside `test/mocks` or `test/units/mocks`.

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 4]
```diff
@@ -280,4 +288,114 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
         );
     }
+
+    // ============ _credit ============
+    function test_credit_SuccessfulMintToUser() public {
+        // Test successful minting to a regular user address
+        uint256 initialBalance = rlcCrosschainToken.balanceOf(user2);
+
+        // Expect the Transfer & CrosschainMint event
+        vm.expectEmit(true, true, true, true);
```
Comment (zguesmi) [inline test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:298]:
```suggestion
        vm.expectEmit(true, true, true, true, address(rlcCrosschainToken));
```

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 5]
```diff
@@ -280,4 +288,114 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+        emit IERC20.Transfer(address(0), user2, TRANSFER_AMOUNT);
+        vm.expectEmit(true, true, true, true);
```
Comment (zguesmi) [inline test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:300]:
```suggestion
        vm.expectEmit(true, true, true, true, address(iexecLayerZeroBridgeChainX));
```

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 6]
```diff
@@ -280,4 +288,114 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+
+        assertEq(amountReceived, TRANSFER_AMOUNT, "Amount received should equal mint amount");
+        assertEq(
+            rlcCrosschainToken.balanceOf(user2),
+            initialBalance + TRANSFER_AMOUNT,
+            "User balance should increase by mint amount"
+        );
+    }
+
+    function testFuzz_credit_Address(address to) public {
```
Comment (zguesmi) [inline test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:313]: This should be:
```suggestion
    function test_credit_SendToDeadAddressInsteadOfZeroAddress(address to) public {
```
No need for fuzzing, just check that the addresse zero is replaced by `0xdead` and check balances.

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 7]
```diff
@@ -280,4 +288,114 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+
+        // Additional check for zero address case
+        if (to == address(0)) {
+            assertEq(rlcCrosschainToken.balanceOf(address(0)), 0, "Zero address balance should remain zero");
+        }
+    }
+
+    function testFuzz_credit_Amount(uint256 amount) public {
+        // Fuzz test with different amounts for testing edge case (0 & max RLC supply)
+        vm.assume(amount <= INITIAL_BALANCE);
```
Comment (zguesmi) [inline test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:335]: I don't see how 0 and "Max RLC supply" are used?

File: test/units/utils/TestUtils.sol [hunk 8]
```diff
@@ -10,82 +10,107 @@ import {UUPSProxyDeployer} from "../../../script/lib/UUPSProxyDeployer.sol";
...
+        address initialUpgrader;
+        address initialPauser;
+    }
 
-        // Deploy RLC token mock for L1
-        rlcToken = new RLCMock();
+    // Struct to hold deployment results
+    struct DeploymentResult {
+        IexecLayerZeroBridge iexecLayerZeroBridgeChainWithApproval;
+        IexecLayerZeroBridge iexecLayerZeroBridgeChainWithoutApproval;
```
Comment (zguesmi) [inline test/units/utils/TestUtils.sol:33]:
```suggestion
        IexecLayerZeroBridge iexecLayerZeroBridgeWithApproval;
        IexecLayerZeroBridge iexecLayerZeroBridgeWithoutApproval;
```

File: test/units/utils/TestUtils.sol [hunk 9]
```diff
@@ -10,82 +10,107 @@ import {UUPSProxyDeployer} from "../../../script/lib/UUPSProxyDeployer.sol";
...
-        iexecLayerZeroBridgeChainA = IexecLayerZeroBridge(
+    function _deployBridge(
+        DeploymentParams memory params,
+        bool approvalRequired,
+        address bridgeableToken,
+        address createXFactory,
+        bytes32 salt
+    ) private returns (IexecLayerZeroBridge) {
+        return IexecLayerZeroBridge(
             UUPSProxyDeployer.deployUsingCreateX(
```
Comment (zguesmi) [inline test/units/utils/TestUtils.sol:83]:
```suggestion
            new IexecLayerZeroBridgeDeployScript().deploy(
```

File: src/mocks/IexecLayerZeroBridgeHarness.sol
Comment (Le-Caignec) [reply src/mocks/IexecLayerZeroBridgeHarness.sol:1]: Same answer as in your pull request => https://github.com/iExecBlockchainComputing/rlc-multichain/pull/73#pullrequestreview-2996769292
It's not possible

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol (hunk 6 above)
Comment (Le-Caignec) [reply test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol]: fix Commit: 49f461c4c29debe139faf566038273c174f0efcb

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol (hunk 7 above)
Comment (Le-Caignec) [reply test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol]: We are testing that the _credit function behaves as expected for all possible values in the range from 0 to the maximum RLC supply.
Fix Commit: f3db4883858b97bd382f5aba0156979c11b32b46

File: test/units/utils/TestUtils.sol (hunk 8 above)
Comment (Le-Caignec) [reply test/units/utils/TestUtils.sol]: fix Commit: c5836a959ff4c32e9ab5c0a1f4edee217902db90

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol (hunk 1 above)
Comment (zguesmi) [reply test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol]: We can keep it for now and clean later.

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 10]
```diff
@@ -95,189 +104,296 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+    //     _test_SendToken_WhenOperational(iexecLayerZeroBridgeEthereum, address(rlcToken), true);
+    // }
+
+    // function test_SendToken_WhenOperational_WithoutApproval() public {
+    //     _test_SendToken_WhenOperational(iexecLayerZeroBridgeChainX, address(rlcCrosschainToken), false);
+    // }
+
+    // function _test_SendToken_WhenOperational(
+    //     IexecLayerZeroBridgeHarness iexecLayerZeroBridge,
+    //     address tokenAddress,
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol]: fix

File: test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol [hunk 11]
```diff
@@ -280,4 +288,112 @@ contract IexecLayerZeroBridgeTest is TestHelperOz5 {
...
+        assertEq(
+            rlcCrosschainToken.balanceOf(user2),
+            initialBalance + TRANSFER_AMOUNT,
+            "User balance should increase by mint amount"
+        );
+    }
+
+    function test_credit_SendToDeadAddressInsteadOfZeroAddress() public {
+        // Fuzz test with different addresses including zero address
+        // Handle zero address redirection
```
Comment (zguesmi) [inline test/units/bridges/layerZero/IexecLayerZeroBridge.t.sol:315]:
```suggestion
    function test_credit_SendToDeadAddressInsteadOfZeroAddress() public {
```
